staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_test.go

```diff
@@ -263,6 +264,53 @@ func TestDeleteObject(t *testing.T) {
 	}
 }
 
+func TestDeleteObjectWithConfirm(t *testing.T) {
+	cmdtesting.InitTestErrorHandler(t)
+	_, _, rc := cmdtesting.TestData()
+
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
```

Comment:
I wholeheartedly detest the TestFactory :sob:, since you know what you want to achieve, and the current construction of the code DOES NOT require you to use that factory, how about rewriting this new test only to inject `resource.Result` which is then being processed by `RunDelete`'s invocation to `DeleteResult`. That will make the test smaller, and explicit about what you're testing. 

In a followup we could drop the test factory entirely from here, also now that I'm looking at the code it's also passing `cmdutil.Factory` to `RunDelete` which is bad :sob: - delete deserves some cleanups :disappointed: but that can be done in followup PRs.

Reply:
I fully agree with you. But even changes in this function also requires good amount of tweaking due to the a lot of fields in DeleteOptions need to be initialized, I mostly worry about not catching any regression in interactive flag changes.

 What about I open a followup PR just for the purpose of tweaking and refactoring this unit test file?